pkg/p2p/libp2p/libp2p.go

```diff
@@ -155,7 +155,12 @@ func New(ctx context.Context, signer beecrypto.Signer, networkID uint64, overlay
 		return nil, fmt.Errorf("autonat: %w", err)
 	}
 
-	handshakeService, err := handshake.New(overlay, h.ID(), signer, networkID, o.Logger)
+	underlays, err := buildFullMAs(h.Addrs(), h.ID())
+	if err != nil {
+		return nil, fmt.Errorf("handshake service: %w", err)
```

Comment:
This error annotation is not related to the action that buildFullMAs does.

Reply:
Ah, I thought it might be in the scope of creating a handshake service, but you are right, it's should have a separate error, good catch!